teps/0140-producing-results-in-matrix.md

```diff
@@ -0,0 +1,702 @@
+
+```yaml
+tasks:
+...
+- name: get-platforms
+  taskSpec:
+    results:
+      - name: platforms
+        type: array
+    steps:
+      - name: write-array
+        image: bash:latest
+        script: |
+          #!/usr/bin/env bash
+          echo -n "[\"linux\",\"mac\",\"windows\"]" | tee $(results.platforms.path)
+```
+
+TaskRun Created
+```
+pr-get-platforms - [\"linux\",\"mac\",\"windows\"]
+```
+
+A `Matrix` when fanned out creates multiple `TaskRuns` so for a `Matrix` to emit results, it is unclear how each `Results` would map back to the original matrixed `PipelineTask`. In the example below, the following `Matrix` will produce 9 `TaskRuns` that will each produce a string `Results` named “report-url” for each combination of platform and browser.  Since the order of the `TaskRuns` is predeterministic, it will always produce `Results` in the same order, but we won’t know how to map each combination `result` to the original params.
+```yaml
+Tasks:
+...
+kind: Task
+metadata:
+  name: taskwithresults
+spec:
+  params:
+    - name: platform
+      default: ""
+    - name: browser
+      default: ""
+  results:
+    - name: report-url
+      type: string
+  steps:
+    - name: produce-report-url
+      image: alpine
+      script: |
+        #!/usr/bin/env bash
+        echo "https://api.example/get-report/$(params.platform)-$(params.browser)" | tee $(results.report-url.path)
+---
+kind: PipelineRun
+metadata:
+  generateName: platforms-with-results
+spec:
+  serviceAccountName: "default"
+  pipelineSpec:
+    tasks:
+      - name: matrix-emitting-results
+        matrix:
+          params:
+            - name: platform
+              value:
+                - linux
+                - mac
+                - windows
+            - name: browser
+              value:
+                - chrome
+                - safari
+                - firefox
+        taskRef:
+          name: taskwithresults
+          kind: Task
+```
+
+TaskRuns Created - “report-url” Result Produced
+```
+pr-produce-platforms-and-browsers-0 - “path/to/report/linux-chrome”
+pr-produce-platforms-and-browsers-1 - “path/to/report/mac-chrome”
+pr-produce-platforms-and-browsers-2 - “path/to/report/windows-chrome”
+pr-produce-platforms-and-browsers-3 - “path/to/report/linux-safari”
+pr-produce-platforms-and-browsers-4 - “path/to/report/mac-safari”
+pr-produce-platforms-and-browsers-5 - “path/to/report/windows-safari”
+pr-produce-platforms-and-browsers-6 - “path/to/report/linux-firefox”
+pr-produce-platforms-and-browsers-7 - “path/to/report/mac-firefox”
+pr-produce-platforms-and-browsers-8 - “path/to/report/windows-firefox”
+```
+
+### Goals
+
+The main goal of this TEP is to enable executing a `Matrixed` `PipelineTask` that can produce `Results`.
+
+### Non-Goals
+
+The following are out of scope for this TEP:
+
+1. Support a matrixed `Results` that produces `Results` of type array or object that are consumed by another `Results`.
```

Comment:
consumed by another `Result` - does this mean you can pass this result using a parameter to another task or something else?

Reply:
This was a typo - supposed to be another `PipelineTask`. Updated.